### DISCUSSION TITLE
Why can the API take a few seconds sometimes?

### DISCUSSION THREAD

**Discussion by Shom770** [weather-gov/api#458] [Q&A]
I'm running API requests for this API asynchronously, it's only around 2 requests yet it is still taking ages after I do the first request, would any of you know why that is? Usually, most APIs don't buckle under 2 asynchronous requests, so I'm wondering if it's on the API's end or on my end.

Thanks!

**Comment by JJKraw** [comment]
I have a cron job that pulls alerts at the state (area) level, 10 at a time, and yes, sometimes the response takes a few seconds. Not being able to control that is the reason I use the cron job. When a user requests the data, it's sitting in a cache file on my server.

**Reply by Shom770** [reply]
> I have a cron job that pulls alerts at the state (area) level, 10 at a time, and yes, sometimes the response takes a few seconds. Not being able to control that is the reason I use the cron job. When a user requests the data, it's sitting in a cache file on my server.

What is a cronjob? Will it prevent the few seconds idling of time?

**Reply by JJKraw** [reply]
Sorry, should have been more clear. it's a feature of the unix / linux operating system that is usually available to web developers by providers who run linux servers. It's a way to have code to run on a schedule totally independent of web activity. So it's not applicable to a lot of other development environments.

**Reply by Shom770** [reply]
> Sorry, should have been more clear. it's a feature of the unix / linux operating system that is usually available to web developers by providers who run linux servers. It's a way to have code to run on a schedule totally independent of web activity. So it's not applicable to a lot of other development environments.

So will caching improve my speed? And how do I cache

**Reply by JJKraw** [reply]
This is getting kind of off the forum focus of the API itself, but what I do is a) fetch the alerts for all regional states at a regular interval (every five minutes); b) write the json that I get back for each state to a file; c) reference that file when I get a user (web page) request rather than do the API fetch.

**Comment by StephenClouse** [comment]
If you are seeing this slowness consistently, please report it to NCO/OMB as an operational issue.
